LibGit2/Functions/Merge-GitCommit.ps1

```diff
@@ -0,0 +1,103 @@
+# limitations under the License.
+
+function Merge-GitCommit
+{
+    <#
+    .SYNOPSIS
+    Merges a commit into the current branch.
+
+    .DESCRIPTION
+    The `Merge-GitCommit` function merges a commit into the current branch. The commit can be identified with its ID, by a tag name, or branch name. It returns a `LibGit2.Automation.MergeResult` object, which has two properties:
+
+    * `Status`: the status of the merge. It will be one of the following values:
+      * `Conflicts`: when there are conflicts with the merge.
+      * `FastForward`: when the merge resulted in a fast-forward.
+      * `NonFastForward`: when a merge commit was created.
+      * `UpToDate`: when nothing needed to be merged.
+    * `Commit`: the merge commit (if one was created).
+
+    If there are conflicts, the conflicts are left in place. You can use your preferred merge tool to resolve the conflicts and then commit. If this script is running non-interactively, you probably don't want any conflict markers hanging out in your local files. Use the "-NonInteractive" switch to prevent conflict files from remaining.
+
+    By default, the function operates on the Git repository in the current directory. Use the `RepoRoot` parameter to target a different repository.
+
+    .EXAMPLE
+    Merge-GitCommit -Revision 'develop'
+
+    Demonstrates how to merge a branch into the current branch.
+    #>
+    [CmdletBinding()]
+    [OutputType([LibGit2.Automation.MergeResult])]
+    param(
+        [string]
+        # The path to the repository where the files should be added. The default is the current directory as returned by Get-Location.
+        $RepoRoot = (Get-Location).ProviderPath,
+
+        [Parameter(Mandatory=$true)]
+        [string]
+        # The revision the tag should point to/reference. A revision can be a specific commit ID/sha (short or long), branch name, tag name, etc. Run git help gitrevisions or go to https://git-scm.com/docs/gitrevisions for full documentation on Git's revision syntax.
+        $Revision = "HEAD",
+
+        [string]
+        [ValidateSet('No','Only')]
+        # Controls whether or not to do a fast-forward merge. By default, "Merge-GitCommit" will try to do a fast-forward merge if it can. If it can't it will create a new merge commit. Options are:
+        # 
+        # * `No`: Don't do a fast-forward merge. Always create a merge commit.
+        # * `Only`: Only do a fast-forward merge. No new merge commit is created.
+        $FastForward,
+
+        [Switch]
+        # The merge is happening non-interactively. If there are any conflicts, the working directory will be left in the state it was in before the merge, i.e. there will be no conflict markers left in any files.
+        $NonInteractive
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    $repo = Get-GitRepository -RepoRoot $RepoRoot

```

Comment:
Should we add some logic to ensure that the current checked out commit is actually a branch so that we're not merging into a detached HEAD?

Reply:
If someone wants to do that, let them. I can see really edge cases where someone wants to do that.